src/service/extensions-impl.js

```diff
@@ -271,27 +265,21 @@ export class Extensions {
    * @param {string} extensionId
    * @param {boolean=} includeInserted If true, includes script elements that
    *   are inserted by the runtime dynamically. Default is true.
-   * @return {?{el: !Element, attr: (string|undefined)}}
+   * @return {?Element}
    * @private
    */
   getExtensionScript_(extensionId, includeInserted = true) {
     const attr = this.attributeForExtension_(extensionId);
-    const modifier = includeInserted ? '' : ':not([i-amphtml-inserted])';
+    // Always ignore <script> elements that have a mismatched RTV.
+    const modifier =
+      ':not([i-amphtml-loaded-new-version])' +
+      (includeInserted ? '' : ':not([i-amphtml-inserted])');
```

Comment:
I don't understand why `includeInserted` is necessary. The only caller is `reloadExtension`? Is it trying to guard against a second call? (It won't work, the second call is guaranteed to throw _because_ of this logic).

Reply:
Two callers:
- `preloadExtension` wants `includeInserted == true` to avoid preloading inserted extensions.
- `reloadExtension` wants `includeInserted == false` to avoid reloading an inserted extension. The throw in this case is WAI.

Reply:
If we were to never add the `:not([i-amphtml-inserted])`, what effectively changes? I think both cases will continue to work.

Reply:
We can ask the person who added it. #24402 😋 Might want to keep and fix though, since we delete the extension holder on the second call and I'm not sure that's okay. 

Anyways this is tangential since this PR preserves status quo. So we (you) can fix it later. :)